algebird-shapeless/src/main/scala/com/twitter/algebird/shapeless/EquivOrdering.scala

```diff
@@ -0,0 +1,40 @@
+package com.twitter.algebird.shapeless
+
+import shapeless._
+
+object EquivOrdering extends EquivOrdering1 {
+  implicit def hconsOrdering[A, B <: HList](implicit
+    la: Lazy[Ordering[A]],
+    b: Ordering[B]): Ordering[A :: B] =
```

Comment:
yeah, not really sure that is needed at all... probably insufficiently reviewed internally.

Should we add the Lazy if it is not needed? I really don't know when it is (e.g. it must not be here, since we are not using it on the tail).

Reply:
@oscar I think this might be causing the failures on 2.10? Checking now…

Reply:
Hmm, swapping the `Lazy` from heads to tails everywhere doesn't fix 2.10. Taking a look now.,

Reply:
Removing the `Lazy` altogether makes at least the `Semigroup` one compile. My guess is some lingering version of [this bug](https://github.com/milessabin/shapeless/issues/367)—I can try to minimize this weekend and open a Shapeless issue.